Added shortcut key for item hide mode

AMartinie opened the pull request:
• Created a shortcut key (default unbinded) that will allow users to toggle item hide/edit mode using a keybind.
• Added more consistency from Hide Mode declarations being used interchangably with Edit Mode.
  ◘Some of this caused a couple booleans to become unused. This is solved in this commit. 
 
Fixes #1529

AMartinie committed:
Added shortcut key for item hide mode

• Created a shortcut key (default unbinded) that will allow users to toggle item hide/edit mode using a keybind.
• Added more consistency from Hide Mode declarations being used interchangably with Edit Mode.
  ◘Some of this caused a couple booleans to become unused. This is solved in this commit.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/mezz/JustEnoughItems?pullRequest=1530) <br/>All committers have signed the CLA.

mezz commented on src/main/java/mezz/jei/gui/overlay/IngredientListOverlay.java:
```diff
@@ -304,10 +304,18 @@ public boolean onGlobalKeyPressed(int eventKey) {
 				setKeyboardFocus(true);
 				return true;
 			}
+			if(KeyBindings.toggleEditMode.isActiveAndMatches(eventKey)) {
```
I think this second check is not needed. If it is, it should have a explanatory comment.

AMartinie replied:
Fixed in [18243e31](https://github.com/mezz/JustEnoughItems/pull/1530/commits/18243e31cfb436d520ca9bceb2fe57c84aee58e7)

mezz commented on src/main/java/mezz/jei/gui/overlay/IngredientListOverlay.java:
```diff
@@ -304,10 +304,18 @@ public boolean onGlobalKeyPressed(int eventKey) {
 				Config.toggleCheatItemsEnabled();
 				return true;
 			}
+			if(KeyBindings.toggleEditMode.isActiveAndMatches(eventKey)) {
```
Please keep the `if (` style with a space between `if` and `(`

AMartinie replied:
Fixed in [18243e31](https://github.com/mezz/JustEnoughItems/pull/1530/commits/18243e31cfb436d520ca9bceb2fe57c84aee58e7)

mezz reviewed (changes requested):
Thanks for contributing!
Mostly looks good, just some comments to resolve.

AMartinie committed:
Update IngredientListOverlay.java

• Corrected a couple styling inconsistencies from my last push.
• Removed repetitive if-statement

AMartinie commented:
Changes have been made and are ready for implementation.

mezz commented:
Looks good, thanks!
